test/test_tensor.py

```diff
@@ -421,6 +421,8 @@ def test_reshape(self):
       # cannot reshape from size 0 to size 1
       a = t.reshape(())
 
+  # TODO: should this be allowed?
+  @unittest.expectedFailure
   def test_expand(self):
     t = Tensor.full((3, 2, 0), 12).expand((6, 2, 0))
     assert t.shape == (6, 2, 0)
```

Comment:
should this expand be allowed?
```
In [2]: a = jnp.full((3, 2, 0), 12)
In [3]: expanded = jnp.broadcast_to(a, (6,2,0))
ValueError: Incompatible shapes for broadcasting: (3, 2, 0) and requested shape (6, 2, 0)

In [5]: torch.full((3, 2, 0), 12).expand((6, 2, 0))
RuntimeError: The expanded size of the tensor (6) must match the existing size (3) at non-singleton dimension 0.  Target sizes: [6, 2, 0].  Tensor sizes: [3, 2, 0]
```

Reply:
i don't quite get why jax and torch does not support it, but maybe that's fine. i would what's more natural from the implementation (meaning don't treat it as a special case)